src/libData/AccountStore/AccountStore.cpp

```diff
@@ -606,3 +692,124 @@ bool AccountStore::RevertCommitTemp() {
 }
 
 void AccountStore::NotifyTimeoutTemp() { m_accountStoreTemp.NotifyTimeout(); }
+
+bool AccountStore::GetProof(const Address& address,
+                            const dev::h256& rootHash,
+                            Account& account,
+                            std::set<std::string>& nodes) {
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING, "not lookup node");
+    return false;
+  }
+
+  std::string rawAccountBase;
+
+  dev::h256 t_rootHash = (rootHash == dev::h256()) ? m_prevRoot : rootHash;
+
+  LOG_GENERAL(INFO, "RootHash " << t_rootHash.hex());
+
+  {
+    std::lock(m_mutexTrie, m_mutexDB);
+    std::lock_guard<std::mutex> lock1(m_mutexTrie, std::adopt_lock);
+    std::lock_guard<std::mutex> lock2(m_mutexDB, std::adopt_lock);
+
+    auto t_state = m_state;
+
+    if (t_rootHash != dev::h256()) {
+      try {
+        t_state.setRoot(t_rootHash);
+      } catch (std::exception& e) {
+        LOG_GENERAL(WARNING, "setRoot for " << t_rootHash.hex() << " failed "
+                                            << e.what());
+        return false;
+      }
+    }
+
+    rawAccountBase = t_state.getProof(
+        DataConversion::StringToCharArray(address.hex()), nodes);
+  }
+
+  if (rawAccountBase.empty()) {
+    return false;
+  }
+
+  Account t_account;
+  if (!t_account.DeserializeBase(
+          zbytes(rawAccountBase.begin(), rawAccountBase.end()), 0)) {
+    LOG_GENERAL(WARNING, "Account::DeserializeBase failed");
+    return false;
+  }
+
+  if (t_account.isContract()) {
+    t_account.SetAddress(address);
+  }
+
+  account = std::move(t_account);
+
+  return true;
+}
+
+bool AccountStore::UpdateStateTrie(const Address& address,
+                                            const Account& account) {
```

Comment:
Does it required clang-format?

Reply:
Running clang-format now messes up the whole codebase, probably as a result of it being disabled for a while?